### PR TITLE
feat: pass the build type as a value to build plugins

### PR DESCRIPTION
**Pull request by wkillerud**
This allows plugins like Eik's ESBuild plugin to only apply for client-side code (like this: https://github.com/podium-lib/eik-podlet-server-extension/pull/4).

At the same time, remove Rollup from the build and let ESBuild do the bundling and tree-shaking. It may perform worse in some situations, but the complexity of the build setup is too high, and plugins like Eik would have to supply both an ESBuild _and_ Rollup version.

### CODE REVIEW

**Comment by wkillerud** [comment]
Looking at https://github.com/mischnic/tree-shaking-example, it's not given that Rollup performs better. The outlier is `react-icons`, but given the note below the comparison I figure we can ignore that extreme case.

Another thing we used Rollup for was inlining dynamic imports, but that's part of ESBuild now (https://github.com/evanw/esbuild/releases/tag/v0.19.0)

**Review by digitalsadhu — COMMENTED** [review]
Removing Rollup is probably the best way to go, good stuff.

At first I wasn't too keen on the type server/client as I was in the mindset that we might try to get rid of bundling server side entirely but then I remembered that plugins like the Warp plugin do indeed need to be applied both server and client side. I'm in 2 minds about whether it should be type "SERVER" and "CLIENT". Might be nicer to convert to flags passed to plugins as in something like `isServer` and `isClient` or something.

Final thing, can we PR this against the next branch instead of main? I've got a few changes in there that aren't on main yet and be good to add this.

File: CHANGELOG.md
Comment (wkillerud) [inline CHANGELOG.md:1]: We could perhaps do without this file, since the content is available in the Releases section on GH?

Comment (digitalsadhu) [reply CHANGELOG.md:1]: Value does seems pretty limited and it is a PITA when it comes to merging next/main after a while.

**Review by digitalsadhu — APPROVED** [review]

**Pull request merged**